src/callkit.cs

```diff
@@ -0,0 +1,509 @@
+﻿//
+// CallKit bindings
+//
+// Authors:
+//	Alex Soto  <alex.soto@xamarin.com>
+//
+// Copyright 2016 Xamarin Inc. All rights reserved.
+//
+
+using System;
+using XamCore.AVFoundation;
+using XamCore.CoreFoundation;
+using XamCore.Foundation;
+using XamCore.ObjCRuntime;
+
+// FIXME: This Api is supposed to be available on macOS 10.12
+// per apple headers but it uses types that are only available
+// in iOS filled radar://26786260 with apple
+// https://trello.com/c/afWXDZ3A
+#if !MONOMAC || XAMCORE_2_0
+namespace XamCore.CallKit {
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Introduced (PlatformName.MacOSX, 10, 12)]
+	[Native]
+	public enum CXAuthorizationStatus : nint {
+		NotDetermined = 0,
+		Restricted = 1,
+		Denied = 2,
+		Authorized = 3
+	}
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Introduced (PlatformName.MacOSX, 10, 12)]
+	[Native]
+	public enum CXCallDirectoryEnabledStatus : nint {
+		Unknown = 0,
+		Disabled = 1,
+		Enabled = 2
+	}
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Introduced (PlatformName.MacOSX, 10, 12)]
+	[Native]
+	public enum CXErrorCode : nint {
+		Unknown = 0
+	}
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Introduced (PlatformName.MacOSX, 10, 12)]
+	[Native]
+	public enum CXErrorCodeIncomingCallError : nint {
+		Unknown = 0,
+		NotAuthorized = 1,
+		CallUuidAlreadyExists = 2,
+		FilteredByDoNotDisturb = 3,
+		FilteredByBlockList = 4
+	}
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Introduced (PlatformName.MacOSX, 10, 12)]
+	[Native]
+	public enum CXErrorCodeRequestTransactionError : nint {
+		Unknown = 0,
+		NotAuthorized = 1,
+		UnknownCallProvider = 2,
+		EmptyTransaction = 3,
+		UnknownCallUuid = 4,
+		InvalidAction = 5,
+		MaximumCallGroupsReached = 6
+	}
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Introduced (PlatformName.MacOSX, 10, 12)]
+	[Native]
+	public enum CXErrorCodeCallDirectoryManagerError : nint {
+		Unknown = 0,
+		NoExtensionFound = 1,
+		NoAttachmentFound = 2,
+		LoadingInterrupted = 3,
+		ParsingFailed = 4,
+		MaximumEntriesExceeded = 5,
+		ExtensionDisabled = 6
+	}
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Introduced (PlatformName.MacOSX, 10, 12)]
+	[Native]
+	public enum CXPlayDtmfCallActionType : nint {
+		SingleTone = 1,
+		SoftPause = 2,
+		HardPause = 3
+	}
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Introduced (PlatformName.MacOSX, 10, 12)]
+	[Native]
+	public enum CXCallEndedReason : nint {
+		Failed = 1,
+		RemoteEnded = 2,
+		Unanswered = 3
+	}
+
+	// FIXME: Should we put these constants somewhere else?
+	[Static]
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Introduced (PlatformName.MacOSX, 10, 12)]
+	interface CXErrorDomain {
```

Comment:
Please use the new `[ErrorDomain]` on the `enums` to expose the domains, that makes them easier to find while coding (more uniform).